core/server/common/src/main/java/alluxio/master/journal/AbstractAdvanceThread.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.journal;
+
+import alluxio.ProcessUtils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Abstract helper for tracking and terminating a thread for journal advance task.
+ */
+public abstract class AbstractAdvanceThread extends Thread {
+  private static final Logger LOG = LoggerFactory.getLogger(AbstractAdvanceThread.class);
+
+  @Override
+  public void run() {
+    try {
+      runAdvance();
+    } catch (Exception e) {
+      ProcessUtils.fatalError(LOG, e, "Advance thread is failed.");
```

Comment:
I try to list the workflow:
```
advancing
When BACKUP_ABORT_AFTER_TRANSITION_TIMEOUT_MS exceed
CompletableFuture will be cancel/interrupted
runAdvance will throw runtime exception
here will make it as a fatal error
system exit
```
Is it what's happening?

Reply:
`CompletableFuture` in `BackupWorkerRole#handleRequestMessage` is used to track `AdvanceFuture#waitTermimation`. Cancelling it won't interrupt advancing. 

Also `AdvanceFuture#cancel` itself won't interrupt thread but just mark for termination.

So timing out on advance won't trigger interrupt and will simply call `resume`.Which will cause advancing to finish prematurely.

I can implement `AdvanceFuture#waitTermination(long timeout)` variation to unclutter this piece of code but that itself will be complicated since `AdvanceFuture` can hold multiple `AdvanceFuture`s

Reply:
@LuQQiu Do you have any more concerns around this?